src/base/static/components/organisms/main-map.tsx

```diff
@@ -28,155 +25,143 @@ import { datasetsSelector, datasetsPropType } from "../../state/ducks/datasets";
-  position: "absolute",
-  top: "8px",
-  left: "8px",
-});
-
-// react-map-gl does not export mapboxgl's built-in geolocate control, so we
-// need to build our own here.
-class GeolocateControl extends Component {
-  onClickGeolocateControl = () => {
-    // https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API
-    navigator.geolocation.getCurrentPosition(position => {
-      this.props.updateMapViewport({
-        longitude: position.coords.longitude,
-        latitude: position.coords.latitude,
-      });
-    });
+// TODO: remove this once we remove the Mapseed global:
+declare const MAP_PROVIDER_TOKEN: string;
 
-    // TODO: Geolocate error handling.
-    // TODO: Set zoom on geolocate? Add pulsating marker?
-  };
+const statePropTypes = {
+  activeEditPlaceId: PropTypes.number,
+  filterableLayerGroupsMetadata: PropTypes.arrayOf(filterableLayerGroupPropType)
+    .isRequired,
+  filteredPlaces: placesPropType.isRequired,
+  interactiveLayerIds: PropTypes.arrayOf(PropTypes.string.isRequired)
+    .isRequired,
+  isContentPanelVisible: PropTypes.bool.isRequired,
+  isMapCenterpointVisible: PropTypes.bool.isRequired,
+  isInviteModalOpen: PropTypes.bool.isRequired,
+  mapConfig: mapConfigPropType,
+  mapContainerDimensions: PropTypes.shape({
+    width: PropTypes.number.isRequired,
+    height: PropTypes.number.isRequired,
+  }).isRequired,
+  mapLayerPopupSelector: PropTypes.func.isRequired,
+  mapStyle: mapStylePropType.isRequired,
+  placeFilters: PropTypes.array.isRequired,
+  datasets: datasetsPropType,
+};
 
-  render() {
-    // If geolocation is not supported, do not render the geolocate control.
-    // This mirrors the default mapboxgl behavior.
-    return "geolocation" in navigator ? (
-      <div
-        className="mapboxgl-ctrl mapboxgl-ctrl-group"
-        style={{ marginTop: "8px" }}
-      >
-        <button
-          className="mapboxgl-ctrl-icon mapboxgl-ctrl-geolocate"
-          type="button"
-          title="Geolocate"
-          onClick={this.onClickGeolocateControl}
-        />
-      </div>
-    ) : null;
-  }
-}
+const dispatchPropTypes = {
+  updateFeaturesInGeoJSONSource: PropTypes.func.isRequired,
+  updateLayers: PropTypes.func.isRequired,
+  updateMapContainerDimensions: PropTypes.func.isRequired,
+};
 
-GeolocateControl.propTypes = {
-  updateMapViewport: PropTypes.func.isRequired,
+type StateProps = PropTypes.InferProps<typeof statePropTypes>;
+type DispatchProps = PropTypes.InferProps<typeof dispatchPropTypes>;
+type ParentProps = {
+  isMapDraggedOrZoomed: boolean;
+  mapContainerWidthDeclaration: string;
+  mapContainerHeightDeclaration: string;
+  mapContainerRef: React.RefObject<HTMLElement>;
+  onUpdateInitialMapViewport: Function;
+  onUpdateMapViewport: Function;
+  onUpdateMapDraggedOrZoomed: Function;
+  onUpdateSpotlightMaskVisibility: Function;
+  onUpdateSourceLoadStatus: Function;
+  mapSourcesLoadStatus: any;
+  mapViewport: MapViewport;
```

Comment:
Instead of using `PropTypes.InferProps`, I think defining our types directly in TS is simpler and less overhead, as we are doing here with `ParentProps`.

It's nice to be able to use our existing PropType definitions, but I think we should phase them out in favor of TS types and interfaces, which can be defined in the relevant `.d.ts` files alongside our ducks.